### PR TITLE
refactor: use sync map instead for map string

### PR DESCRIPTION
**Pull request by nerdeveloper**
Signed-off-by: nerdeveloper <odirionye@gmail.com>

### CODE REVIEW

**Comment by nerdeveloper** [comment]
Yes

File: scripts/test.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ mkdir .cover/ .test/
 trap "rm -rf .test/" EXIT
 
 for pkg in `go list ./... | grep -v /vendor/`; do
-    go test -v -covermode=atomic \
+    go test -race -v -covermode=atomic \
```
Comment (cbuto) [inline scripts/test.sh:11]: you'll need this set so the tests run!
```suggestion
    CGO_ENABLED=1 go test -race -v -covermode=atomic \
```

File: pkg/chartmuseum/server/multitenant/server.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ type (
 		StorageBackend         cm_storage.Backend
 		TimestampTolerance     time.Duration
 		ExternalCacheStore     cache.Store
-		InternalCacheStore     map[string]*cacheEntry
+		InternalCacheStore     sync.Map
```
Comment (scbizu) [inline pkg/chartmuseum/server/multitenant/server.go:52]: Can we wrap the `sync.Map` inside a specialized structure , for example `type memoryCacheStore struct { sync.Map }` maybe ,  and wrap the `Load` and `Store` func ?  Doing a type assertion every where  seems ugly though .

**Comment by scbizu** [comment]
@nerdeveloper  And please fix your DCO verification ?

**Review by scbizu — CHANGES_REQUESTED** [review]

**Comment by cbuto** [comment]
@nerdeveloper is there anything we can help with here?

**Comment by nerdeveloper** [comment]
Yes, @cbuto, I tried to refactor to sync map, but the race condition still exists. i am having a hard time debugging this

**Comment by nerdeveloper** [comment]
I am going to close this PR, as @cbuto has picked it and finished it

**Pull request closed without merging**